relay/kusama/src/lib.rs

```diff
@@ -1927,6 +1927,211 @@ pub mod migrations {
 		}
 	}
 
+	/// Migration to remove deprecated judgement proxies.
+	mod clear_judgement_proxies {
+		use super::*;
+
+		use frame_support::{
+			pallet_prelude::ValueQuery,
+			storage_alias,
+			traits::{Currency, ReservableCurrency},
+			Twox64Concat,
+		};
+		use frame_system::pallet_prelude::BlockNumberFor;
+		use pallet_proxy::ProxyDefinition;
+		use sp_runtime::{BoundedVec, Saturating};
+
+		/// ProxyType including the deprecated `IdentityJudgement`.
+		#[derive(
+			Copy,
+			Clone,
+			Eq,
+			PartialEq,
+			Ord,
+			PartialOrd,
+			Encode,
+			Decode,
+			RuntimeDebug,
+			MaxEncodedLen,
+			TypeInfo,
+		)]
+		pub enum PrevProxyType {
+			Any,
+			NonTransfer,
+			Governance,
+			Staking,
+			IdentityJudgement,
+			CancelProxy,
+			Auction,
+			Society,
+			NominationPools,
+		}
+
+		type BalanceOf<T> = <<T as pallet_proxy::Config>::Currency as Currency<
+			<T as frame_system::Config>::AccountId,
+		>>::Balance;
+
+		type PrevProxiesValue<T> = (
+			BoundedVec<ProxyDefinition<AccountId, PrevProxyType, BlockNumberFor<T>>, MaxProxies>,
+			BalanceOf<T>,
+		);
+
+		/// Proxies including the deprecated `IdentityJudgement` type.
+		#[storage_alias]
+		pub type Proxies<T: pallet_proxy::Config> = StorageMap<
+			pallet_proxy::Pallet<T>,
+			Twox64Concat,
+			AccountId,
+			PrevProxiesValue<T>,
+			ValueQuery,
+		>;
+
+		pub struct Migration;
+		impl OnRuntimeUpgrade for Migration {
+			/// Compute the expected post-upgrade state for Proxies stroage, and the reserved value
+			/// for all accounts with a proxy.
+			#[cfg(feature = "try-runtime")]
+			fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+				let mut expected_proxies: BTreeMap<AccountId, PrevProxiesValue<Runtime>> =
+					BTreeMap::new();
+				let mut expected_reserved_amounts: BTreeMap<AccountId, Balance> = BTreeMap::new();
+
+				for (who, (mut proxies, old_deposit)) in
+					Proxies::<Runtime>::iter().collect::<Vec<_>>()
+				{
+					let proxies_len_before = proxies.len() as u64;
+					proxies.retain(|proxy| proxy.proxy_type != PrevProxyType::IdentityJudgement);
+					let proxies_len_after = proxies.len() as u64;
+
+					let new_deposit =
+						pallet_proxy::Pallet::<Runtime>::deposit(proxies.len() as u32);
+
+					let current_reserved =
+						<Balances as ReservableCurrency<AccountId>>::reserved_balance(&who);
+
+					// Update the deposit only if proxies were removed and the deposit decreased.
+					if new_deposit < old_deposit && proxies_len_after < proxies_len_before {
+						// If there're no proxies left, they should be removed
+						if proxies.len() > 0 {
+							expected_proxies.insert(who.clone(), (proxies, new_deposit));
+						}
+						expected_reserved_amounts.insert(
+							who,
+							current_reserved.saturating_sub(old_deposit - new_deposit),
+						);
+					} else {
+						// Nothing should change
+						expected_proxies.insert(who.clone(), (proxies, old_deposit));
+						expected_reserved_amounts.insert(who, current_reserved);
```

Comment:
Is this correct? The above `if` only triggers if *both* were modified. Maybe there needs to be an else if to trigger for either of them changed as well.

Reply:
It's deliberate, we only make changes if both were modified.

Reply:
Lets suppose one proxy was removed, so first the user had 5 and now they have 4. But at the same time the deposit function was increased from 20 to 25. Then the num proxies decreased but the deposit stays the same, or?  
But its a very edge case 😆

Reply:
I think that's fine, in that case we just remove the invalid proxy and leave the deposit at 20.

Now that you mention it though my comment is misleading, since the `proxies` could still have changed. I'll adjust it.

Reply:
Updated the comment.